src/primitives/transaction.h

```diff
@@ -391,23 +496,36 @@ class CTransaction
         bool isOverwinterV3 = fOverwintered &&
                               nVersionGroupId == OVERWINTER_VERSION_GROUP_ID &&
                               nVersion == 3;
-        if (fOverwintered && !isOverwinterV3) {
+        bool isSaplingV4 =
+            fOverwintered &&
+            nVersionGroupId == SAPLING_VERSION_GROUP_ID &&
+            nVersion == SAPLING_TX_VERSION;
```

Comment:
Nit: I'd prefer the indentation (and the use of a number vs ``*_TX_VERSION`` macro) to be consistent between Overwinter and Sapling, here and below, but that can wait to the next PR that touches this code.